ArduCopter/GCS_Mavlink.cpp

```diff
@@ -195,6 +195,16 @@ void GCS_MAVLINK_Copter::send_nav_controller_output() const
 
 float GCS_MAVLINK_Copter::vfr_hud_airspeed() const
 {
+#if AP_AIRSPEED_ENABLED
+    // airspeed sensors are best. While the AHRS airspeed_estimate
+    // will use an airspeed sensor, that value is constrained by the
+    // ground speed. When reporting we should send the true airspeed
+    // value if possible:
+    if (copter.g2.airspeed.enabled() && copter.g2.airspeed.healthy()) {
+        return copter.g2.airspeed.get_airspeed();
+    }
+#endif
+
     Vector3f airspeed_vec_bf;
     if (AP::ahrs().airspeed_vector_true(airspeed_vec_bf)) {
```

Comment:
I just want to point out that the estimated airspeed output by AHRS is a total airspeed meaning it is the length of the airspeed along all three axes.  This is much different than the airspeed provided by the sensor which is along a single axis (x axis).